### PR TITLE
Add transaction timestamp to Subscription output #370

### PR DESCRIPTION
**Pull request by th1nkNXT**
- I assumed that there is always an existing transaction (or unconfirmed transaction) (transactionId = subscriptionId).
- I implemented the additional tx information only for the mentioned endpoint `getSubscription`.
- I introduced a new AccountDto class which is delivered in the JSON data as a possible future improvement of the data format and code.
- That means ALL subscription JSON objects deliver now a `senderAccount` object and a `recipientAccount` object. 
- This data is now redundant to be backward compatible but you could think about that to go for a new API version in the future.
- I wrote a test to ensure that the JSON data is correctly created (and still backward compatible).
- I added Lombok to the maven pom to reduce boilerplate code. That means when integrating my changes you and other devs need to install the Lombok plugin into your IDE. 
- You should then have a look at my code, maybe there are some things to use for future developments (e.g. using a data transfer object class).

example data (old):
`{"id":"9223372036854775807","sender":"123456789","senderRS":"BURST-RMAP-225P-CQQP-22222",																		"recipient":"23456789","recipientRS":"BURST-DV2P-222Q-DTX3-22222",																		   "amountNQT":"321","frequency":4,"timeNext":5}
`example data (new):
`{"id":"9223372036854775807","sender":"123456789","senderRS":"BURST-RMAP-225P-CQQP-22222","senderAccount":{"id":"123456789","idRS":"BURST-RMAP-225P-CQQP-22222"},"recipient":"23456789","recipientRS":"BURST-DV2P-222Q-DTX3-22222","recipientAccount":{"id":"23456789","idRS":"BURST-DV2P-222Q-DTX3-22222"},"amountNQT":"321","frequency":4,"timeNext":5,"timestamp":1623878907,"blockTimestamp":1623878900}
`

### CODE REVIEW

File: src/brs/http/JSONData.java [hunk 1]
```diff
@@ -1,22 +1,129 @@
 package brs.http;
 
-import brs.*;
+import static brs.http.common.ResultFields.ACCOUNT_RESPONSE;
```
Comment (damccull) [inline src/brs/http/JSONData.java:3]: I would revert this bit back to `import brs.*`. I assume this was an IDE change, thinking its helping.

Comment (th1nkNXT) [reply src/brs/http/JSONData.java:3]: Yes, it was an automatic IDE change. Sorry for that. I will revert it.

File: src/brs/http/GetSubscription.java [hunk 2]
```diff
@@ -42,7 +47,9 @@ JsonElement processRequest(HttpServletRequest req) {
       response.addProperty(ERROR_DESCRIPTION_RESPONSE, "Subscription not found");
       return response;
     }
+    // retrieve additional transaction data
```
Comment (ohager) [inline src/brs/http/GetSubscription.java:50]: seems like an not-so-useful comment

File: src/brs/http/JSONData.java [hunk 3]
```diff
@@ -263,14 +267,30 @@ static JsonObject purchase(DigitalGoodsStore.Purchase purchase) {
...
+        .recipientRS(recipient.getIdRS())
+        .recipientAccount(recipient)
+        .amountNQT(Convert.toUnsignedLong(subscription.getAmountNQT()))
+        .frequency(subscription.getFrequency())
+        .timeNext(subscription.getTimeNext())
+        .timestamp(transaction != null ? transaction.getTimestamp() : null)
+        .blockTimestamp(transaction != null ? transaction.getBlockTimestamp() : null)
+        .build());
+    
+    return json.getAsJsonObject();
```
Comment (ohager) [inline src/brs/http/JSONData.java:293]: definitely better readable :+1:

File: src/brs/http/api/SubscriptionDto.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+
+@AllArgsConstructor
+@Builder
+public class SubscriptionDto {
+	public String id;
+
+	// TODO sender and senderRS for backward compatibility.
+	// With the type AccountDto there is a new way to provide the data using more reusable code.
```
Comment (ohager) [inline src/brs/http/api/SubscriptionDto.java:12]: no need for such comments - code is selfexplaining IMHO

**Review by ohager — APPROVED** [review]
LGTM - @jjos2372 for the final verdict :grin:

File: src/brs/http/GetSubscription.java (hunk 2 above)
Comment (th1nkNXT) [reply src/brs/http/GetSubscription.java:50]: Yes, maybe it was more like a thought which i had in that case ;-) i will remove it.

File: src/brs/http/api/SubscriptionDto.java (hunk 4 above)
Comment (th1nkNXT) [reply src/brs/http/api/SubscriptionDto.java]: done

File: src/brs/http/GetSubscription.java (hunk 2 above)
Comment (th1nkNXT) [reply src/brs/http/GetSubscription.java]: done

**Comment by ohager** [comment]
@jjos2372 is this something relevant?! PR is stale already... maybe we can close it

**Comment by frankTheTank72** [comment]
@ohager do we still need this change or will we cover that in a different way - now we have the Openaapi ?

**Comment by frankTheTank72** [comment]
added to 3.5.2 beta2

**Comment by ohager** [comment]
Stale.... and very old! I think we updated that differenlty already after so long time. @jjos2372 Closing?! I recommend. 
> We should be better on reviews by other contributors.

**Comment by ohager** [comment]
Close due to stale. Hurts me a bit to see a PR pending for so much time, and finally not accepting a community members contribution... but bigger changes are coming in. We might review the idea of adding more info to subscriptions.

**Pull request closed without merging**